src/core/Object3D.js

```diff
@@ -28,6 +28,36 @@ const _removedEvent = { type: 'removed' };
 const _childaddedEvent = { type: 'childadded', child: null };
 const _childremovedEvent = { type: 'childremoved', child: null };
 
+function listenToProperties( object, properties, onReadCallback, onWriteCallback ) {
```

Comment:
This could be moved to math classes' getters/setters, but I thought inlining here would make them easier to maintain.

Most of the diff from this PR is removing the indirection and callbacks from math classes.